src/ShopifyApp/Middleware/AuthShop.php

```diff
@@ -48,10 +48,17 @@ protected function validateShop(Request $request)
         $session = new ShopSession();
 
         // Grab the shop's myshopify domain from query or session
-        $shopDomainParam = $request->get('shop');
+        // For SPA's we need X-Shop-Domain : See issue https://github.com/ohmybrew/laravel-shopify/issues/295
+        $shopDomainParam = $request->get('shop') ?? $request->header('X-Shop-Domain');
         $shopDomainSession = $session->getDomain();
         $shopDomain = ShopifyApp::sanitizeShopDomain($shopDomainParam ?? $shopDomainSession);
 
+        // See issue https://github.com/ohmybrew/laravel-shopify/issues/295
+        parse_str(parse_url($request->header('referer'), PHP_URL_QUERY), $refererQueryParams);
+        if (isset($refererQueryParams['shop']) && $shopDomain !== $refererQueryParams['shop'] && ShopifyApp::api()->verifyRequest($refererQueryParams)) {
+            $shopDomain = $refererQueryParams['shop'];
+        }
+
```

Comment:
You shouldn't combine PRs like this.

1 feature/fix per PR should be followed.

This now contains my semi-unrelated referer fix as well as a change to allow modifying of store to be loaded via request headers.

Reply:
Please remove this functionality from this PR as it exists in another PR. Leave fixes where they belongs - in their own PRs.

If changes are made in either PR then both would be out of sync.

For example, I am busy adding tests to make sure coverage in my PR goes up.

We should also be having the conversation about re-verification on the PR where this code is.

Thanks! 😎

Reply:
I have completely refactored my PR and this is now completely different.

Please remove all of these old lines of mine from your PR.

Thank you!